apps/dc_tools/odc/apps/dc_tools/index_from_tar.py

```diff
@@ -1,13 +1,143 @@
 """ Index datasets from tar arachive
 """
 
+import os
 import sys
+from copy import deepcopy
+
 import click
 import datacube
 from datacube.utils.changes import allow_any
+from odc.index import from_yaml_doc_stream, odc_uuid
 from odc.io.tar import tar_doc_stream, tar_mode
 from odc.io.timer import RateEstimator
-from odc.index import from_yaml_doc_stream
+from pyproj import CRS, Transformer
+
+
+KNOWN_CONSTELLATIONS = [
+    'sentinel-2'
+]
+
+
+def _stac_product_lookup(item):
+    product_label = item['id']
+    product_name = item['properties']['platform']
+    region_code = None
+
+    constellation = item['properties']['constellation']
+
+    if constellation in KNOWN_CONSTELLATIONS:
+        if constellation == 'sentinel-2':
+            product_label = item['properties']["sentinel:product_id"]
+            product_split = product_label.split("_")
+            product_name = f"{product_split[0]}_{product_split[1]}"
+            region_code = "{}{}{}".format(
+                str(item['properties']["proj:epsg"])[-2:],
+                item['properties']["sentinel:latitude_band"],
+                item['properties']["sentinel:grid_square"]
+            )
+
+    return product_label, product_name, region_code
+
+
+def _get_stac_bands(item, default_grid='g10.0m'):
+    bands = {}
+
+    grids = {}
+
+    assets = item['assets']
+
+    for asset_name, asset in assets.items():
+        # Ignore items that are not actual COGs
+        if asset['type'] not in ["image/tiff; application=geotiff; profile=cloud-optimized"]:
+            continue
+
+        transform = asset['proj:transform']
+        grid = "g{}m".format(transform[0])
+
+        if grid not in grids:
+            grids[grid] = {
+                'shape': asset['proj:shape'],
+                'transform': asset['proj:transform']
+            }
+
+        band_info = {
+            'path': os.path.split((asset['href']))[1]
+        }
+
+        if grid != default_grid:
+            band_info['grid'] = grid
+
+        bands[asset_name] = band_info
+
+    grids['default'] = grids[default_grid]
+    del grids[default_grid]
+
+    return bands, grids
+
+
+def geographic_to_projected(geometry, target_srs):
```

Comment:
Is this funtion in use? Which GDAL this is supposed to work with? lon/lat order will be wrong for one of them